guides/05_chatbots/02_chat_interface_examples.md

```diff
@@ -1,144 +1,57 @@
-In this Guide, we go through several examples of how to use `gr.ChatInterface` with popular LLM libraries.
+In this Guide, we go through several examples of how to use `gr.ChatInterface` with popular LLM libraries and API providers.
 
-## A `langchain` example
+We will cover the following libraries and API providers:
 
-Let's start by using `langchain` on top of `openai` to build a general-purpose streaming chatbot application in 19 lines of code. You'll need to have an OpenAI key for this example (keep reading for the free, open-source equivalent!)
+* [Llama Index](#llama-index)
+* [LangChain](#langchain)
+* [OpenAI](#openai)
+* [Hugging Face `transformers`](#hugging-face-transformers)
+* [SambaNova](#sambanova)
+* [Hyperbolic](#hyperbolic)
 
-```python
-from langchain.chat_models import ChatOpenAI
-from langchain.schema import AIMessage, HumanMessage
-import openai
-import gradio as gr
+For many LLM libraries and providers, there exist community-maintained integration libraries that make it even easier to spin up Gradio apps. We reference these libraries in the appropriate sections below.
 
-os.environ["OPENAI_API_KEY"] = "sk-..."  # Replace with your key
+## Llama Index
 
-llm = ChatOpenAI(temperature=1.0, model='gpt-3.5-turbo-0613')
+Let's start by using `llama-index` on top of `openai` to build a RAG chatbot on any text or PDF files that you can demo and share in less than 30 lines of code. You'll need to have an OpenAI key for this example (keep reading for the free, open-source equivalent!)
 
-def predict(message, history):
-    history_langchain_format = []
-    for msg in history:
-        if msg['role'] == "user":
-            history_langchain_format.append(HumanMessage(content=msg['content']))
-        elif msg['role'] == "assistant":
-            history_langchain_format.append(AIMessage(content=msg['content']))
-    history_langchain_format.append(HumanMessage(content=message))
-    gpt_response = llm(history_langchain_format)
-    return gpt_response.content
+$code_llm_llamaindex
 
-gr.ChatInterface(predict, type="messages").launch()
-```
+## LangChain
 
-## A streaming example using `openai`
+Here's an example using `langchain` on top of `openai` to build a general-purpose chatbot. As before, you'll need to have an OpenAI key for this example.
 
-Of course, we could also use the `openai` library directy. Here a similar example, but this time with streaming results as well:
+$code_llm_langchain
 
-```python
-from openai import OpenAI
-import gradio as gr
+**Note**: For quick prototyping, the community-maintained [langchain-gradio repo](https://github.com/AK391/langchain-gradio)  makes it even easier to build chatbots on top of LangChain.
```

Comment:
Why not use 'Tip' for these notes to make them more prominent? unfortunately there's an open issue with tip and markdown links. will make a fix for that but for now you'll need to use <a> tags.

Reply:
```suggestion
Tip: For quick prototyping, the community-maintained <a href='https://github.com/AK391/langchain-gradio'>langchain-gradio repo</a>  makes it even easier to build chatbots on top of LangChain.
```

Reply:
That's exactly the reason why I didn't do Tip, but if it supports html tags, then this sounds good 👍